docs_src/design/adr/0023-North-South-Messaging.md

```diff
@@ -0,0 +1,313 @@
+    AuthMode = "usernamepassword"                   # required for redis messagebus (secure or insecure).
+    SecretName = "redisdb"
+    [ExternalMessageQueue]
+    Protocol = "tcp"
+    Host = "localhost"
+    Port = 6378
+    Type = "mqtt"
+    RequestTopic = "my-app/command/request/#"           # for subscribing to 3rd party command requests
+    ResponseTopicPrefix = "my-app/command/response/"    # for publishing responses back to 3rd party systems /<device-name>/<command-name>/<method> will be added to this publish topic prefix
+```
+
+!!! Note
+        Core command contains no MessageQueue configuration today.  This is all additive/new configuration and therefore backward compatible with EdgeX 2.x implementations.
+
+#### device service configuration
+
+Example device service configuration is provided below.
+
+``` toml
+[MessageQueue]
+## already existing message queue configuration (for sending events/readings to the message bus)
+Protocol = "redis"
+Host = "localhost"
+Port = 6379
+Type = "redis"
+AuthMode = "usernamepassword"                           # required for redis messagebus (secure or insecure).
+SecretName = "redisdb"
+PublishTopicPrefix = "edgex/events/device" # /<device-profile-name>/<device-name>/<source-name> will be added to this Publish Topic prefix
+  [MessageQueue.Optional]
+  # Default MQTT Specific options that need to be here to enable environment variable overrides of them
+  # Client Identifiers
+  ClientId = "device-rest"
+  # Connection information
+  Qos = "0" # Quality of Sevice values are 0 (At most once), 1 (At least once) or 2 (Exactly once)
+  KeepAlive = "10" # Seconds (must be 2 or greater)
+  Retained = "false"
+  AutoReconnect = "true"
+  ConnectTimeout = "5" # Seconds
+  SkipCertVerify = "false" # Only used if Cert/Key file or Cert/Key PEMblock are specified
+
+## new configuration to allow device services to also communicate via message bus with core command
+CommandRequestTopic = "edgex/command/request/#"         # subscribing for inbound command requests
+CommandResponseTopicPrefix = "edgex/command/response/"  # publishing outbound command responses; <device-service>/<device-name>/<command-name>/<method> will be added to this publish topic prefix
+```
+
+!!! Note
+        Most of the device service configuration is existing based on its need to already communicate with the message bus for publishing events/readings.  The last two lines are added to allow device services to subscribe and publish command messages from/to the message bus.
+
+## Questions
+
+- Do we need separate topics for all the devices or would one on the device service suffice?
+    
+    - Ans:  we have defined the deviceName in the parameterized topic, so one topic should be sufficient for Device Service- edgex/command/request/<device-name>...
+
+- Would clients (non EdgeX services and applications) want to get a list of available commands via message (instead of calling REST)?
```

Comment:
I would think it's important that we provide parity with the existing REST implementation. If we don't provide a means to get the command list via message bus, then we force 3rd party applications to use both REST and message bus.

Reply:
Added sections to cover queries via message bus